// @mistralai/mistralai/esm/lib/config.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 320761608fb3
 */
import { pathToFunc } from "./url.js";
/**
 * Global Production server
 */
export const ServerGlobal = "global";
/**
 * EU Production server
 */
export const ServerEu = "eu";
/**
 * US Production server
 */
export const ServerUs = "us";
/**
 * Contains the list of servers available to the SDK
 */
export const ServerList = {
    [ServerGlobal]: "https://api.mistral.ai",
    [ServerEu]: "https://api.eu.mistral.ai",
    [ServerUs]: "https://api.us.mistral.ai",
};
export function serverURLFromOptions(options) {
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const server = options.server ?? ServerGlobal;
        serverURL = ServerList[server] || "";
    }
    const u = pathToFunc(serverURL)(params);
    return new URL(u);
}
export const SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "1.0.0",
    sdkVersion: "2.5.0",
    genVersion: "2.884.13",
    userAgent: "speakeasy-sdk/typescript 2.5.0 2.884.13 1.0.0 @mistralai/mistralai",
};
