// @mistralai/mistralai/esm/funcs/betaRagSearchIndexesGenerateSchemaSummary.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: afee370f33eb
 */
import * as z from "zod/v4";
import { encodeSimple } from "../lib/encodings.js";
import { matchStatusCode } from "../lib/http.js";
import { JsonLStream } from "../lib/jsonl.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as errors from "../models/errors/index.js";
import * as operations from "../models/operations/index.js";
import { APIPromise } from "../types/async.js";
/**
 * Generate a summary field for a schema
 *
 * @remarks
 * Streams a summary for the schema in chunks of json.
 *
 * The first chunk contains metadata for the summary, the following contain
 * chunks of 'content' that should be joined together to form a full summary.
 */
export function betaRagSearchIndexesGenerateSchemaSummary(client, request, options) {
    return new APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = safeParse(request, (value) => operations
        .GenerateSchemaSummaryPostV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePostRequest$outboundSchema
        .parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [parsed, { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = null;
    const pathParams = {
        index_id: encodeSimple("index_id", payload.index_id, {
            explode: false,
            charEncoding: "percent",
        }),
        language: encodeSimple("language", payload.language, {
            explode: false,
            charEncoding: "percent",
        }),
        schema_id: encodeSimple("schema_id", payload.schema_id, {
            explode: false,
            charEncoding: "percent",
        }),
    };
    const path = pathToFunc("/v1/rag/indexes/index/{index_id}/schemas/schema/{schema_id}/summary_field/{language}")(pathParams);
    const headers = new Headers(compactMap({
        Accept: "application/x-ndjson",
    }));
    const secConfig = await extractSecurity(client._options.apiKey);
    const securityInput = secConfig == null ? {} : { apiKey: secConfig };
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "generate_schema_summary_post_v1_rag_indexes_index__index_id__schemas_schema__schema_id__summary_field__language__post",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: client._options.apiKey,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "POST",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || 300000,
    }, options);
    if (!requestRes.ok) {
        return [requestRes, { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        isErrorStatusCode: (statusCode) => matchStatusCode({ status: statusCode }, ["4XX", "5XX"]),
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [doResult, { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result] = await M.match(M.jsonl(200, z.custom(x => x instanceof ReadableStream)
        .transform(stream => {
        return new JsonLStream({
            stream,
            decoder(rawEvent) {
                const schema = operations
                    .GenerateSchemaSummaryPostV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePostSummaryStreamTypes$inboundSchema;
                return schema.parse(rawEvent);
            },
        });
    }), { ctype: "application/x-ndjson" }), M.jsonErr(422, errors.HTTPValidationError$inboundSchema), M.fail([400, 403, 404, "4XX"]), M.fail([500, "5XX"]))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [result, { status: "complete", request: req, response }];
    }
    return [result, { status: "complete", request: req, response }];
}
