// @mistralai/mistralai/esm/funcs/betaPromptsList.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: ee315b088fa3
 */
import { dlv } from "../lib/dlv.js";
import { encodeFormQuery } from "../lib/encodings.js";
import { matchStatusCode } from "../lib/http.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import * as operations from "../models/operations/index.js";
import { APIPromise } from "../types/async.js";
import { createPageIterator, haltIterator, } from "../types/operations.js";
/**
 * ListPrompts
 */
export function betaPromptsList(client, request, options) {
    return new APIPromise($do(client, request, options));
}
async function $do(client, request, options) {
    const parsed = safeParse(request, (value) => operations.PromptsListRequest$outboundSchema.optional().parse(value), "Input validation failed");
    if (!parsed.ok) {
        return [haltIterator(parsed), { status: "invalid" }];
    }
    const payload = parsed.value;
    const body = null;
    const path = pathToFunc("/v2/prompts")();
    const query = encodeFormQuery({
        "alias": payload?.alias,
        "fields": payload?.fields,
        "pageSize": payload?.pageSize,
        "pageToken": payload?.pageToken,
    });
    const headers = new Headers(compactMap({
        Accept: "application/json",
    }));
    const secConfig = await extractSecurity(client._options.apiKey);
    const securityInput = secConfig == null ? {} : { apiKey: secConfig };
    const requestSecurity = resolveGlobalSecurity(securityInput);
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "prompts_list",
        oAuth2Scopes: null,
        resolvedSecurity: requestSecurity,
        securitySource: client._options.apiKey,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        security: requestSecurity,
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        query: query,
        body: body,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || 300000,
    }, options);
    if (!requestRes.ok) {
        return [haltIterator(requestRes), { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        isErrorStatusCode: (statusCode) => matchStatusCode({ status: statusCode }, ["4XX", "5XX"]),
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [haltIterator(doResult), { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const responseFields = {
        HttpMeta: { Response: response, Request: req },
    };
    const [result, raw] = await M.match(M.json(200, operations.PromptsListResponse$inboundSchema, {
        key: "Result",
    }), M.fail("4XX"), M.fail("5XX"))(response, req, { extraFields: responseFields });
    if (!result.ok) {
        return [haltIterator(result), {
                status: "complete",
                request: req,
                response,
            }];
    }
    const nextFunc = (responseData) => {
        const nextCursor = dlv(responseData, "nextPageToken");
        if (typeof nextCursor !== "string") {
            return { next: () => null };
        }
        if (nextCursor.trim() === "") {
            return { next: () => null };
        }
        const results = dlv(responseData, "data");
        if (!Array.isArray(results) || !results.length) {
            return { next: () => null };
        }
        const limit = request?.pageSize ?? 0;
        if (results.length < limit) {
            return { next: () => null };
        }
        const nextVal = () => betaPromptsList(client, {
            ...request,
            pageToken: nextCursor,
        }, options);
        return { next: nextVal, "~next": { cursor: nextCursor } };
    };
    const page = { ...result, ...nextFunc(raw) };
    return [{ ...page, ...createPageIterator(page, (v) => !v.ok) }, {
            status: "complete",
            request: req,
            response,
        }];
}
