// @mistralai/mistralai/esm/models/operations/agentsapiv1agentslistpages.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 895c67a63ffd
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as components from "../components/index.js";
/** @internal */
export const AgentsApiV1AgentsListPagesRequest$outboundSchema = z.object({
    pageSize: z.int().default(20),
    deploymentChat: z.nullable(z.boolean()).optional(),
    sources: z.nullable(z.array(components.RequestSource$outboundSchema))
        .optional(),
    name: z.nullable(z.string()).optional(),
    search: z.nullable(z.string()).optional(),
    id: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.string(), z.any())).optional(),
    pageToken: z.nullable(z.string()).optional(),
}).transform((v) => {
    return remap$(v, {
        pageSize: "page_size",
        deploymentChat: "deployment_chat",
        pageToken: "page_token",
    });
});
export function agentsApiV1AgentsListPagesRequestToJSON(agentsApiV1AgentsListPagesRequest) {
    return JSON.stringify(AgentsApiV1AgentsListPagesRequest$outboundSchema.parse(agentsApiV1AgentsListPagesRequest));
}
/** @internal */
export const AgentsApiV1AgentsListPagesResponse$inboundSchema = z.object({
    Result: components.AgentListPage$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function agentsApiV1AgentsListPagesResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AgentsApiV1AgentsListPagesResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AgentsApiV1AgentsListPagesResponse' from JSON`);
}
