// @mistralai/mistralai/esm/models/operations/generateindexsummaryv1ragindexesindexindexidsummaryfieldlanguagepost.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 71ab6a31fe57
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
import * as components from "../components/index.js";
export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage = {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
};
/** @internal */
export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage$outboundSchema = z.enum(GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage);
/** @internal */
export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest$outboundSchema = z.object({
    indexId: z.string(),
    language: GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        indexId: "index_id",
    });
});
export function generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequestToJSON(generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest) {
    return JSON.stringify(GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest$outboundSchema
        .parse(generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest));
}
/** @internal */
export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes$inboundSchema = smartUnion([
    components.SummaryMetadata$inboundSchema,
    components.SummaryChunk$inboundSchema,
    components.SummaryError$inboundSchema,
]);
export function generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes' from JSON`);
}
