// @mistralai/mistralai/esm/models/operations/getindexsummaryv1ragindexesindexindexidsummaryfieldlanguageget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0b8d6ed58e09
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
export const GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage = {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
};
/** @internal */
export const GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage$outboundSchema = z.enum(GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage);
/** @internal */
export const GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest$outboundSchema = z.object({
    indexId: z.string(),
    language: GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        indexId: "index_id",
    });
});
export function getIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequestToJSON(getIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest) {
    return JSON.stringify(GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest$outboundSchema
        .parse(getIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest));
}
