// @mistralai/mistralai/esm/models/operations/getschemasummaryv1ragindexesindexindexidschemasschemaschemaidsummaryfieldlanguageget.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: fc1c3a7b9822
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
export const GetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetLanguage = {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
};
/** @internal */
export const GetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetLanguage$outboundSchema = z.enum(GetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetLanguage);
/** @internal */
export const GetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetRequest$outboundSchema = z.object({
    indexId: z.string(),
    schemaId: z.string(),
    language: GetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetLanguage$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        indexId: "index_id",
        schemaId: "schema_id",
    });
});
export function getSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetRequestToJSON(getSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetRequest) {
    return JSON.stringify(GetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetRequest$outboundSchema
        .parse(getSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguageGetRequest));
}
