// @mistralai/mistralai/esm/models/operations/promptsupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: cfdc043733bf
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";
/** @internal */
export const UpdatePromptRequest$outboundSchema = z.object({
    title: z.nullable(z.string()).optional(),
    description: z.nullable(z.string()).optional(),
    sharingScope: components.RegistrySharingScope$outboundSchema.optional(),
});
export function updatePromptRequestToJSON(updatePromptRequest) {
    return JSON.stringify(UpdatePromptRequest$outboundSchema.parse(updatePromptRequest));
}
/** @internal */
export const PromptsUpdateRequest$outboundSchema = z.object({
    promptId: z.string(),
    requestBody: z.lazy(() => UpdatePromptRequest$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        promptId: "prompt_id",
        requestBody: "RequestBody",
    });
});
export function promptsUpdateRequestToJSON(promptsUpdateRequest) {
    return JSON.stringify(PromptsUpdateRequest$outboundSchema.parse(promptsUpdateRequest));
}
