// @mistralai/mistralai/esm/models/operations/promptsupdateversionmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f6b0b69eb311
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";
/** @internal */
export const UpdatePromptVersionRequest$outboundSchema = z.object({
    notes: z.nullable(z.string()).optional(),
    aliases: components.AliasList$outboundSchema.optional(),
});
export function updatePromptVersionRequestToJSON(updatePromptVersionRequest) {
    return JSON.stringify(UpdatePromptVersionRequest$outboundSchema.parse(updatePromptVersionRequest));
}
/** @internal */
export const PromptsUpdateVersionMetadataRequest$outboundSchema = z.object({
    promptId: z.string(),
    version: z.int(),
    requestBody: z.lazy(() => UpdatePromptVersionRequest$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        promptId: "prompt_id",
        requestBody: "RequestBody",
    });
});
export function promptsUpdateVersionMetadataRequestToJSON(promptsUpdateVersionMetadataRequest) {
    return JSON.stringify(PromptsUpdateVersionMetadataRequest$outboundSchema.parse(promptsUpdateVersionMetadataRequest));
}
