// @mistralai/mistralai/esm/models/operations/promptscreateversion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: cdbfae10ac5f
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";
/** @internal */
export const CreatePromptVersionRequest$outboundSchema = z.object({
    definition: components.PromptDefinition$outboundSchema,
    notes: z.nullable(z.string()).optional(),
    aliases: z.array(z.string()).optional(),
});
export function createPromptVersionRequestToJSON(createPromptVersionRequest) {
    return JSON.stringify(CreatePromptVersionRequest$outboundSchema.parse(createPromptVersionRequest));
}
/** @internal */
export const PromptsCreateVersionRequest$outboundSchema = z.object({
    promptId: z.string(),
    requestBody: z.lazy(() => CreatePromptVersionRequest$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        promptId: "prompt_id",
        requestBody: "RequestBody",
    });
});
export function promptsCreateVersionRequestToJSON(promptsCreateVersionRequest) {
    return JSON.stringify(PromptsCreateVersionRequest$outboundSchema.parse(promptsCreateVersionRequest));
}
