// @mistralai/mistralai/esm/models/operations/promptslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7e6054cd6360
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as components from "../components/index.js";
/** @internal */
export const PromptsListRequest$outboundSchema = z.object({
    pageSize: z.int().optional(),
    pageToken: z.string().optional(),
    alias: z.string().optional(),
    fields: z.array(z.string()).optional(),
});
export function promptsListRequestToJSON(promptsListRequest) {
    return JSON.stringify(PromptsListRequest$outboundSchema.parse(promptsListRequest));
}
/** @internal */
export const PromptsListResponse$inboundSchema = z.object({
    Result: components.ListPromptsResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function promptsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PromptsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PromptsListResponse' from JSON`);
}
