// @mistralai/mistralai/esm/models/operations/setindexsummaryv1ragindexesindexindexidsummaryfieldlanguageput.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 053edbeba171
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";
export const SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage = {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
};
/** @internal */
export const SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage$outboundSchema = z.enum(SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage);
/** @internal */
export const SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest$outboundSchema = z.object({
    indexId: z.string(),
    language: SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage$outboundSchema,
    updateSummaryRequestSummary: components.UpdateSummaryRequestSummary$outboundSchema,
}).transform((v) => {
    return remap$(v, {
        indexId: "index_id",
        updateSummaryRequestSummary: "UpdateSummaryRequestSummary",
    });
});
export function setIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequestToJSON(setIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest) {
    return JSON.stringify(SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest$outboundSchema
        .parse(setIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest));
}
