// @mistralai/mistralai/esm/models/operations/skillslist.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3eff96e8881c
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import * as components from "../components/index.js";
/** @internal */
export const SkillsListRequest$outboundSchema = z.object({
    pageSize: z.int().optional(),
    pageToken: z.string().optional(),
    alias: z.string().optional(),
    fields: z.array(z.string()).optional(),
});
export function skillsListRequestToJSON(skillsListRequest) {
    return JSON.stringify(SkillsListRequest$outboundSchema.parse(skillsListRequest));
}
/** @internal */
export const SkillsListResponse$inboundSchema = z.object({
    Result: components.ListSkillsResponse$inboundSchema,
}).transform((v) => {
    return remap$(v, {
        "Result": "result",
    });
});
export function skillsListResponseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SkillsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SkillsListResponse' from JSON`);
}
