// @mistralai/mistralai/esm/models/operations/skillscreateversion.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e89ac7211d41
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";
/** @internal */
export const CreateSkillVersionRequest$outboundSchema = z.object({
    definition: components.SkillDefinition$outboundSchema,
    notes: z.nullable(z.string()).optional(),
    aliases: z.array(z.string()).optional(),
});
export function createSkillVersionRequestToJSON(createSkillVersionRequest) {
    return JSON.stringify(CreateSkillVersionRequest$outboundSchema.parse(createSkillVersionRequest));
}
/** @internal */
export const SkillsCreateVersionRequest$outboundSchema = z.object({
    skillId: z.string(),
    requestBody: z.lazy(() => CreateSkillVersionRequest$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        skillId: "skill_id",
        requestBody: "RequestBody",
    });
});
export function skillsCreateVersionRequestToJSON(skillsCreateVersionRequest) {
    return JSON.stringify(SkillsCreateVersionRequest$outboundSchema.parse(skillsCreateVersionRequest));
}
