// @mistralai/mistralai/esm/models/operations/skillsupdate.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3511172f0c03
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";
/** @internal */
export const UpdateSkillRequest$outboundSchema = z.object({
    sharingScope: components.RegistrySharingScope$outboundSchema.optional(),
});
export function updateSkillRequestToJSON(updateSkillRequest) {
    return JSON.stringify(UpdateSkillRequest$outboundSchema.parse(updateSkillRequest));
}
/** @internal */
export const SkillsUpdateRequest$outboundSchema = z.object({
    skillId: z.string(),
    requestBody: z.lazy(() => UpdateSkillRequest$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        skillId: "skill_id",
        requestBody: "RequestBody",
    });
});
export function skillsUpdateRequestToJSON(skillsUpdateRequest) {
    return JSON.stringify(SkillsUpdateRequest$outboundSchema.parse(skillsUpdateRequest));
}
