// @mistralai/mistralai/esm/models/operations/skillsupdateversionmetadata.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e11580c09cf3
 */
import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";
/** @internal */
export const UpdateSkillVersionRequest$outboundSchema = z.object({
    notes: z.nullable(z.string()).optional(),
    aliases: components.AliasList$outboundSchema.optional(),
});
export function updateSkillVersionRequestToJSON(updateSkillVersionRequest) {
    return JSON.stringify(UpdateSkillVersionRequest$outboundSchema.parse(updateSkillVersionRequest));
}
/** @internal */
export const SkillsUpdateVersionMetadataRequest$outboundSchema = z.object({
    skillId: z.string(),
    version: z.int(),
    requestBody: z.lazy(() => UpdateSkillVersionRequest$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        skillId: "skill_id",
        requestBody: "RequestBody",
    });
});
export function skillsUpdateVersionMetadataRequestToJSON(skillsUpdateVersionMetadataRequest) {
    return JSON.stringify(SkillsUpdateVersionMetadataRequest$outboundSchema.parse(skillsUpdateVersionMetadataRequest));
}
