// @mistralai/mistralai/src/models/components/createpromptrequest.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8a9e4285e88d
 */

import * as z from "zod/v4";
import {
  PromptDefinition,
  PromptDefinition$Outbound,
  PromptDefinition$outboundSchema,
} from "./promptdefinition.js";
import {
  RegistrySharingScope,
  RegistrySharingScope$outboundSchema,
} from "./registrysharingscope.js";

export type CreatePromptRequest = {
  /**
   * Stable object name.
   */
  name: string;
  /**
   * Versioned prompt content.
   */
  definition: PromptDefinition;
  /**
   * Display title.
   */
  title?: string | null | undefined;
  /**
   * Display description.
   */
  description?: string | null | undefined;
  /**
   * Notes for this version.
   */
  notes?: string | null | undefined;
  sharingScope?: RegistrySharingScope | undefined;
  /**
   * Aliases pointing to this version.
   */
  aliases?: Array<string> | undefined;
};

/** @internal */
export type CreatePromptRequest$Outbound = {
  name: string;
  definition: PromptDefinition$Outbound;
  title?: string | null | undefined;
  description?: string | null | undefined;
  notes?: string | null | undefined;
  sharingScope?: string | undefined;
  aliases?: Array<string> | undefined;
};

/** @internal */
export const CreatePromptRequest$outboundSchema: z.ZodType<
  CreatePromptRequest$Outbound,
  CreatePromptRequest
> = z.object({
  name: z.string(),
  definition: PromptDefinition$outboundSchema,
  title: z.nullable(z.string()).optional(),
  description: z.nullable(z.string()).optional(),
  notes: z.nullable(z.string()).optional(),
  sharingScope: RegistrySharingScope$outboundSchema.optional(),
  aliases: z.array(z.string()).optional(),
});

export function createPromptRequestToJSON(
  createPromptRequest: CreatePromptRequest,
): string {
  return JSON.stringify(
    CreatePromptRequest$outboundSchema.parse(createPromptRequest),
  );
}
