// @mistralai/mistralai/src/models/components/getsummaryresponsesummary.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0b23ba39be20
 */

import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import { SummaryStatus, SummaryStatus$inboundSchema } from "./summarystatus.js";

export type GetSummaryResponseSummary = {
  content: string;
  status: SummaryStatus;
  translated: boolean;
};

/** @internal */
export const GetSummaryResponseSummary$inboundSchema: z.ZodType<
  GetSummaryResponseSummary,
  unknown
> = z.object({
  content: z.string(),
  status: SummaryStatus$inboundSchema,
  translated: z.boolean(),
});

export function getSummaryResponseSummaryFromJSON(
  jsonString: string,
): SafeParseResult<GetSummaryResponseSummary, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => GetSummaryResponseSummary$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'GetSummaryResponseSummary' from JSON`,
  );
}
