// @mistralai/mistralai/src/models/components/listpromptsresponse.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 94ff95af1bf8
 */

import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import { Prompt, Prompt$inboundSchema } from "./prompt.js";

export type ListPromptsResponse = {
  data?: Array<Prompt> | undefined;
  nextPageToken?: string | undefined;
};

/** @internal */
export const ListPromptsResponse$inboundSchema: z.ZodType<
  ListPromptsResponse,
  unknown
> = z.object({
  data: z.array(Prompt$inboundSchema).optional(),
  nextPageToken: z.string().optional(),
});

export function listPromptsResponseFromJSON(
  jsonString: string,
): SafeParseResult<ListPromptsResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => ListPromptsResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'ListPromptsResponse' from JSON`,
  );
}
