// @mistralai/mistralai/src/models/components/prompt.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: bffa6c75bb04
 */

import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  PromptDefinition,
  PromptDefinition$inboundSchema,
} from "./promptdefinition.js";
import {
  RegistrySharingScope,
  RegistrySharingScope$inboundSchema,
} from "./registrysharingscope.js";

export type Prompt = {
  id?: string | undefined;
  /**
   * Stable object name.
   */
  name?: string | undefined;
  /**
   * Versioned prompt content.
   */
  definition?: PromptDefinition | undefined;
  version?: number | undefined;
  /**
   * Notes for this version.
   */
  notes?: string | undefined;
  /**
   * Aliases pointing to this version.
   */
  aliases?: Array<string> | undefined;
  sharingScope?: RegistrySharingScope | undefined;
  /**
   * RFC 3339 timestamp.
   */
  createdAt?: Date | undefined;
  /**
   * RFC 3339 timestamp.
   */
  updatedAt?: Date | undefined;
  /**
   * Latest version number.
   */
  latestVersion?: number | undefined;
  /**
   * Display title.
   */
  title?: string | undefined;
  /**
   * Display description.
   */
  description?: string | undefined;
};

/** @internal */
export const Prompt$inboundSchema: z.ZodType<Prompt, unknown> = z.object({
  id: z.string().optional(),
  name: z.string().optional(),
  definition: PromptDefinition$inboundSchema.optional(),
  version: z.int().optional(),
  notes: z.string().optional(),
  aliases: z.array(z.string()).optional(),
  sharingScope: RegistrySharingScope$inboundSchema.optional(),
  createdAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
    .optional(),
  updatedAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
    .optional(),
  latestVersion: z.int().optional(),
  title: z.string().optional(),
  description: z.string().optional(),
});

export function promptFromJSON(
  jsonString: string,
): SafeParseResult<Prompt, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => Prompt$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'Prompt' from JSON`,
  );
}
