// @mistralai/mistralai/src/models/components/promptversion.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1f2518b3cccb
 */

import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  PromptDefinition,
  PromptDefinition$inboundSchema,
} from "./promptdefinition.js";

export type PromptVersion = {
  version?: number | undefined;
  /**
   * Versioned prompt content.
   */
  definition?: PromptDefinition | undefined;
  /**
   * Notes for this version.
   */
  notes?: string | undefined;
  /**
   * Aliases pointing to this version.
   */
  aliases?: Array<string> | undefined;
  /**
   * RFC 3339 timestamp.
   */
  createdAt?: Date | undefined;
};

/** @internal */
export const PromptVersion$inboundSchema: z.ZodType<PromptVersion, unknown> = z
  .object({
    version: z.int().optional(),
    definition: PromptDefinition$inboundSchema.optional(),
    notes: z.string().optional(),
    aliases: z.array(z.string()).optional(),
    createdAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
      .optional(),
  });

export function promptVersionFromJSON(
  jsonString: string,
): SafeParseResult<PromptVersion, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => PromptVersion$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'PromptVersion' from JSON`,
  );
}
