// @mistralai/mistralai/src/models/components/skillversion.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: c602e6218b20
 */

import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  SkillDefinition,
  SkillDefinition$inboundSchema,
} from "./skilldefinition.js";

export type SkillVersion = {
  version?: number | undefined;
  /**
   * Versioned skill content.
   */
  definition?: SkillDefinition | undefined;
  /**
   * Notes for this version.
   */
  notes?: string | undefined;
  /**
   * Aliases pointing to this version.
   */
  aliases?: Array<string> | undefined;
  /**
   * RFC 3339 timestamp.
   */
  createdAt?: Date | undefined;
};

/** @internal */
export const SkillVersion$inboundSchema: z.ZodType<SkillVersion, unknown> = z
  .object({
    version: z.int().optional(),
    definition: SkillDefinition$inboundSchema.optional(),
    notes: z.string().optional(),
    aliases: z.array(z.string()).optional(),
    createdAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
      .optional(),
  });

export function skillVersionFromJSON(
  jsonString: string,
): SafeParseResult<SkillVersion, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => SkillVersion$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'SkillVersion' from JSON`,
  );
}
