// @mistralai/mistralai/src/models/components/summarymetadata.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 377e322f69a3
 */

import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";
import {
  SummaryStatusDollarDefs,
  SummaryStatusDollarDefs$inboundSchema,
} from "./summarystatusdollardefs.js";

/**
 * Summary content. Expected to be the first value returned.
 */
export type SummaryMetadata = {
  status: SummaryStatusDollarDefs;
  translated: boolean;
};

/** @internal */
export const SummaryMetadata$inboundSchema: z.ZodType<
  SummaryMetadata,
  unknown
> = z.object({
  status: SummaryStatusDollarDefs$inboundSchema,
  translated: z.boolean(),
});

export function summaryMetadataFromJSON(
  jsonString: string,
): SafeParseResult<SummaryMetadata, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => SummaryMetadata$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'SummaryMetadata' from JSON`,
  );
}
