// @mistralai/mistralai/src/models/components/updatesummaryrequestsummary.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e503e00d86eb
 */

import * as z from "zod/v4";
import { ClosedEnum } from "../../types/enums.js";

export const UpdateSummaryRequestSummaryStatus = {
  Handwritten: "handwritten",
  GeneratedConfirmed: "generated_confirmed",
} as const;
export type UpdateSummaryRequestSummaryStatus = ClosedEnum<
  typeof UpdateSummaryRequestSummaryStatus
>;

export type UpdateSummaryRequestSummary = {
  content: string;
  status: UpdateSummaryRequestSummaryStatus;
  translated: boolean;
};

/** @internal */
export const UpdateSummaryRequestSummaryStatus$outboundSchema: z.ZodEnum<
  typeof UpdateSummaryRequestSummaryStatus
> = z.enum(UpdateSummaryRequestSummaryStatus);

/** @internal */
export type UpdateSummaryRequestSummary$Outbound = {
  content: string;
  status: string;
  translated: boolean;
};

/** @internal */
export const UpdateSummaryRequestSummary$outboundSchema: z.ZodType<
  UpdateSummaryRequestSummary$Outbound,
  UpdateSummaryRequestSummary
> = z.object({
  content: z.string(),
  status: UpdateSummaryRequestSummaryStatus$outboundSchema,
  translated: z.boolean(),
});

export function updateSummaryRequestSummaryToJSON(
  updateSummaryRequestSummary: UpdateSummaryRequestSummary,
): string {
  return JSON.stringify(
    UpdateSummaryRequestSummary$outboundSchema.parse(
      updateSummaryRequestSummary,
    ),
  );
}
