// @mistralai/mistralai/src/models/components/useridentity.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 80e3ee18f008
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type UserIdentity = {
  id: string;
  email: string | null;
  firstName: string | null;
  lastName: string | null;
};

/** @internal */
export const UserIdentity$inboundSchema: z.ZodType<UserIdentity, unknown> = z
  .object({
    id: z.string(),
    email: z.nullable(z.string()),
    first_name: z.nullable(z.string()),
    last_name: z.nullable(z.string()),
  }).transform((v) => {
    return remap$(v, {
      "first_name": "firstName",
      "last_name": "lastName",
    });
  });

export function userIdentityFromJSON(
  jsonString: string,
): SafeParseResult<UserIdentity, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => UserIdentity$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'UserIdentity' from JSON`,
  );
}
