// @mistralai/mistralai/esm/sdk/beta.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: afeac65e28f8
 */
import { ClientSDK } from "../lib/sdks.js";
import { BetaAgents } from "./betaagents.js";
import { Connectors } from "./connectors.js";
import { Conversations } from "./conversations.js";
import { Libraries } from "./libraries.js";
import { Observability } from "./observability.js";
import { Prompts } from "./prompts.js";
import { Rag } from "./rag.js";
import { Skills } from "./skills.js";
import { Users } from "./users.js";
export class Beta extends ClientSDK {
    _prompts;
    get prompts() {
        return (this._prompts ?? (this._prompts = new Prompts(this._options)));
    }
    _skills;
    get skills() {
        return (this._skills ?? (this._skills = new Skills(this._options)));
    }
    _conversations;
    get conversations() {
        return (this._conversations ?? (this._conversations = new Conversations(this._options)));
    }
    _agents;
    get agents() {
        return (this._agents ?? (this._agents = new BetaAgents(this._options)));
    }
    _libraries;
    get libraries() {
        return (this._libraries ?? (this._libraries = new Libraries(this._options)));
    }
    _observability;
    get observability() {
        return (this._observability ?? (this._observability = new Observability(this._options)));
    }
    _connectors;
    get connectors() {
        return (this._connectors ?? (this._connectors = new Connectors(this._options)));
    }
    _rag;
    get rag() {
        return (this._rag ?? (this._rag = new Rag(this._options)));
    }
    _users;
    get users() {
        return (this._users ?? (this._users = new Users(this._options)));
    }
}
