// @mistralai/mistralai/esm/sdk/prompts.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a20685c0ab9f
 */
import { betaPromptsCreate } from "../funcs/betaPromptsCreate.js";
import { betaPromptsCreateVersion } from "../funcs/betaPromptsCreateVersion.js";
import { betaPromptsDelete } from "../funcs/betaPromptsDelete.js";
import { betaPromptsGet } from "../funcs/betaPromptsGet.js";
import { betaPromptsGetVersion } from "../funcs/betaPromptsGetVersion.js";
import { betaPromptsList } from "../funcs/betaPromptsList.js";
import { betaPromptsListVersions } from "../funcs/betaPromptsListVersions.js";
import { betaPromptsUpdateMetadata } from "../funcs/betaPromptsUpdateMetadata.js";
import { betaPromptsUpdateVersionMetadata } from "../funcs/betaPromptsUpdateVersionMetadata.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class Prompts extends ClientSDK {
    /**
     * ListPrompts
     */
    async list(request, options) {
        return unwrapResultIterator(betaPromptsList(this, request, options));
    }
    /**
     * CreatePrompt
     */
    async create(request, options) {
        return unwrapAsync(betaPromptsCreate(this, request, options));
    }
    /**
     * GetPrompt
     */
    async get(request, options) {
        return unwrapAsync(betaPromptsGet(this, request, options));
    }
    /**
     * DeletePrompt
     */
    async delete(request, options) {
        return unwrapAsync(betaPromptsDelete(this, request, options));
    }
    /**
     * UpdatePrompt
     */
    async updateMetadata(promptId, requestBody, options) {
        return unwrapAsync(betaPromptsUpdateMetadata(this, promptId, requestBody, options));
    }
    /**
     * ListPromptVersions
     */
    async listVersions(request, options) {
        return unwrapAsync(betaPromptsListVersions(this, request, options));
    }
    /**
     * CreatePromptVersion
     */
    async createVersion(promptId, requestBody, options) {
        return unwrapAsync(betaPromptsCreateVersion(this, promptId, requestBody, options));
    }
    /**
     * GetPromptVersion
     */
    async getVersion(request, options) {
        return unwrapAsync(betaPromptsGetVersion(this, request, options));
    }
    /**
     * UpdatePromptVersionMetadata
     */
    async updateVersionMetadata(promptId, version, requestBody, options) {
        return unwrapAsync(betaPromptsUpdateVersionMetadata(this, promptId, version, requestBody, options));
    }
}
