// @mistralai/mistralai/esm/sdk/searchindexes.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 9a1b36425210
 */
import { betaRagSearchIndexesDocumentLookup } from "../funcs/betaRagSearchIndexesDocumentLookup.js";
import { betaRagSearchIndexesDocumentsFetch } from "../funcs/betaRagSearchIndexesDocumentsFetch.js";
import { betaRagSearchIndexesGenerateIndexSummary } from "../funcs/betaRagSearchIndexesGenerateIndexSummary.js";
import { betaRagSearchIndexesGenerateSchemaSummary } from "../funcs/betaRagSearchIndexesGenerateSchemaSummary.js";
import { betaRagSearchIndexesGetIndexDetail } from "../funcs/betaRagSearchIndexesGetIndexDetail.js";
import { betaRagSearchIndexesGetIndexSchemaDetail } from "../funcs/betaRagSearchIndexesGetIndexSchemaDetail.js";
import { betaRagSearchIndexesGetIndexSchemaFile } from "../funcs/betaRagSearchIndexesGetIndexSchemaFile.js";
import { betaRagSearchIndexesGetIndexSummaries } from "../funcs/betaRagSearchIndexesGetIndexSummaries.js";
import { betaRagSearchIndexesGetIndexSummary } from "../funcs/betaRagSearchIndexesGetIndexSummary.js";
import { betaRagSearchIndexesGetSchemaSummary } from "../funcs/betaRagSearchIndexesGetSchemaSummary.js";
import { betaRagSearchIndexesRegister } from "../funcs/betaRagSearchIndexesRegister.js";
import { betaRagSearchIndexesSetIndexSummary } from "../funcs/betaRagSearchIndexesSetIndexSummary.js";
import { betaRagSearchIndexesSetSchemaSummary } from "../funcs/betaRagSearchIndexesSetSchemaSummary.js";
import { betaRagSearchIndexesUnregister } from "../funcs/betaRagSearchIndexesUnregister.js";
import { betaRagSearchIndexesUpdateIndexMetrics } from "../funcs/betaRagSearchIndexesUpdateIndexMetrics.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
export class SearchIndexes extends ClientSDK {
    /**
     * Register (or re-register) a search index
     */
    async register(request, options) {
        return unwrapAsync(betaRagSearchIndexesRegister(this, request, options));
    }
    /**
     * Get Index Summaries
     *
     * @remarks
     * Fetch summary view of all indexes available to a user
     */
    async getIndexSummaries(options) {
        return unwrapAsync(betaRagSearchIndexesGetIndexSummaries(this, options));
    }
    /**
     * Unregister Search Index
     *
     * @remarks
     * Delete all information about an index
     */
    async unregister(request, options) {
        return unwrapAsync(betaRagSearchIndexesUnregister(this, request, options));
    }
    /**
     * Update Index Metrics
     *
     * @remarks
     * Update the metrics for a given index
     */
    async updateIndexMetrics(request, options) {
        return unwrapAsync(betaRagSearchIndexesUpdateIndexMetrics(this, request, options));
    }
    /**
     * Get Index Details
     *
     * @remarks
     * Get a detailed view of the stored data for a single index
     */
    async getIndexDetail(request, options) {
        return unwrapAsync(betaRagSearchIndexesGetIndexDetail(this, request, options));
    }
    /**
     * Get Index Summary
     *
     * @remarks
     * Retrieve the summary field for an index if it exists
     */
    async getIndexSummary(request, options) {
        return unwrapAsync(betaRagSearchIndexesGetIndexSummary(this, request, options));
    }
    /**
     * Generate a summary field for an index
     *
     * @remarks
     * Streams a summary for the index in chunks of json.
     *
     * The first chunk contains metadata for the summary, the following contain
     * chunks of 'content' that should be joined together to form a full summary.
     */
    async generateIndexSummary(request, options) {
        return unwrapAsync(betaRagSearchIndexesGenerateIndexSummary(this, request, options));
    }
    /**
     * Set Index Summary
     *
     * @remarks
     * Update the summary field for an index
     */
    async setIndexSummary(request, options) {
        return unwrapAsync(betaRagSearchIndexesSetIndexSummary(this, request, options));
    }
    /**
     * Get Schema Summary
     *
     * @remarks
     * Retrieve the summary field for a schema if it exists
     */
    async getSchemaSummary(request, options) {
        return unwrapAsync(betaRagSearchIndexesGetSchemaSummary(this, request, options));
    }
    /**
     * Generate a summary field for a schema
     *
     * @remarks
     * Streams a summary for the schema in chunks of json.
     *
     * The first chunk contains metadata for the summary, the following contain
     * chunks of 'content' that should be joined together to form a full summary.
     */
    async generateSchemaSummary(request, options) {
        return unwrapAsync(betaRagSearchIndexesGenerateSchemaSummary(this, request, options));
    }
    /**
     * Set Schema Summary
     *
     * @remarks
     * Update the summary field for an index
     */
    async setSchemaSummary(request, options) {
        return unwrapAsync(betaRagSearchIndexesSetSchemaSummary(this, request, options));
    }
    /**
     * Get Index Schema Detail
     *
     * @remarks
     * Get a detailed view of the stored information for a schema
     */
    async getIndexSchemaDetail(request, options) {
        return unwrapAsync(betaRagSearchIndexesGetIndexSchemaDetail(this, request, options));
    }
    /**
     * Get Index Schema File
     */
    async getIndexSchemaFile(request, options) {
        return unwrapAsync(betaRagSearchIndexesGetIndexSchemaFile(this, request, options));
    }
    /**
     * Document Lookup
     *
     * @remarks
     * Fetch stored information about a retrievable element stored in an index
     */
    async documentLookup(request, options) {
        return unwrapAsync(betaRagSearchIndexesDocumentLookup(this, request, options));
    }
    /**
     * Document Fetch
     *
     * @remarks
     * Fetch a few stored retrievable elements from the index/schema
     */
    async documentsFetch(request, options) {
        return unwrapAsync(betaRagSearchIndexesDocumentsFetch(this, request, options));
    }
}
