// @mistralai/mistralai/esm/sdk/skills.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 847667a72f83
 */
import { betaSkillsCreate } from "../funcs/betaSkillsCreate.js";
import { betaSkillsCreateVersion } from "../funcs/betaSkillsCreateVersion.js";
import { betaSkillsDelete } from "../funcs/betaSkillsDelete.js";
import { betaSkillsGet } from "../funcs/betaSkillsGet.js";
import { betaSkillsGetVersion } from "../funcs/betaSkillsGetVersion.js";
import { betaSkillsList } from "../funcs/betaSkillsList.js";
import { betaSkillsListVersions } from "../funcs/betaSkillsListVersions.js";
import { betaSkillsUpdateMetadata } from "../funcs/betaSkillsUpdateMetadata.js";
import { betaSkillsUpdateVersionMetadata } from "../funcs/betaSkillsUpdateVersionMetadata.js";
import { ClientSDK } from "../lib/sdks.js";
import { unwrapAsync } from "../types/fp.js";
import { unwrapResultIterator } from "../types/operations.js";
export class Skills extends ClientSDK {
    /**
     * ListSkills
     */
    async list(request, options) {
        return unwrapResultIterator(betaSkillsList(this, request, options));
    }
    /**
     * CreateSkill
     */
    async create(request, options) {
        return unwrapAsync(betaSkillsCreate(this, request, options));
    }
    /**
     * GetSkill
     */
    async get(request, options) {
        return unwrapAsync(betaSkillsGet(this, request, options));
    }
    /**
     * DeleteSkill
     */
    async delete(request, options) {
        return unwrapAsync(betaSkillsDelete(this, request, options));
    }
    /**
     * UpdateSkill
     */
    async updateMetadata(skillId, requestBody, options) {
        return unwrapAsync(betaSkillsUpdateMetadata(this, skillId, requestBody, options));
    }
    /**
     * ListSkillVersions
     */
    async listVersions(request, options) {
        return unwrapAsync(betaSkillsListVersions(this, request, options));
    }
    /**
     * CreateSkillVersion
     */
    async createVersion(skillId, requestBody, options) {
        return unwrapAsync(betaSkillsCreateVersion(this, skillId, requestBody, options));
    }
    /**
     * GetSkillVersion
     */
    async getVersion(request, options) {
        return unwrapAsync(betaSkillsGetVersion(this, request, options));
    }
    /**
     * UpdateSkillVersionMetadata
     */
    async updateVersionMetadata(skillId, version, requestBody, options) {
        return unwrapAsync(betaSkillsUpdateVersionMetadata(this, skillId, version, requestBody, options));
    }
}
