// @mistralai/mistralai/src/funcs/betaRagSearchIndexesGenerateIndexSummary.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 68e65ad818c3
 */

import * as z from "zod/v4";
import { MistralCore } from "../core.js";
import { encodeSimple } from "../lib/encodings.js";
import { matchStatusCode } from "../lib/http.js";
import { JsonLStream } from "../lib/jsonl.js";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { RequestOptions } from "../lib/sdks.js";
import { extractSecurity, resolveGlobalSecurity } from "../lib/security.js";
import { pathToFunc } from "../lib/url.js";
import {
  ConnectionError,
  InvalidRequestError,
  RequestAbortedError,
  RequestTimeoutError,
  UnexpectedClientError,
} from "../models/errors/httpclienterrors.js";
import * as errors from "../models/errors/index.js";
import { MistralError } from "../models/errors/mistralerror.js";
import { ResponseValidationError } from "../models/errors/responsevalidationerror.js";
import { SDKValidationError } from "../models/errors/sdkvalidationerror.js";
import * as operations from "../models/operations/index.js";
import { APICall, APIPromise } from "../types/async.js";
import { Result } from "../types/fp.js";

/**
 * Generate a summary field for an index
 *
 * @remarks
 * Streams a summary for the index in chunks of json.
 *
 * The first chunk contains metadata for the summary, the following contain
 * chunks of 'content' that should be joined together to form a full summary.
 */
export function betaRagSearchIndexesGenerateIndexSummary(
  client: MistralCore,
  request:
    operations.GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest,
  options?: RequestOptions,
): APIPromise<
  Result<
    JsonLStream<
      operations.GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes
    >,
    | errors.HTTPValidationError
    | MistralError
    | ResponseValidationError
    | ConnectionError
    | RequestAbortedError
    | RequestTimeoutError
    | InvalidRequestError
    | UnexpectedClientError
    | SDKValidationError
  >
> {
  return new APIPromise($do(
    client,
    request,
    options,
  ));
}

async function $do(
  client: MistralCore,
  request:
    operations.GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest,
  options?: RequestOptions,
): Promise<
  [
    Result<
      JsonLStream<
        operations.GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes
      >,
      | errors.HTTPValidationError
      | MistralError
      | ResponseValidationError
      | ConnectionError
      | RequestAbortedError
      | RequestTimeoutError
      | InvalidRequestError
      | UnexpectedClientError
      | SDKValidationError
    >,
    APICall,
  ]
> {
  const parsed = safeParse(
    request,
    (value) =>
      operations
        .GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest$outboundSchema
        .parse(value),
    "Input validation failed",
  );
  if (!parsed.ok) {
    return [parsed, { status: "invalid" }];
  }
  const payload = parsed.value;
  const body = null;

  const pathParams = {
    index_id: encodeSimple("index_id", payload.index_id, {
      explode: false,
      charEncoding: "percent",
    }),
    language: encodeSimple("language", payload.language, {
      explode: false,
      charEncoding: "percent",
    }),
  };
  const path = pathToFunc(
    "/v1/rag/indexes/index/{index_id}/summary_field/{language}",
  )(pathParams);

  const headers = new Headers(compactMap({
    Accept: "application/x-ndjson",
  }));

  const secConfig = await extractSecurity(client._options.apiKey);
  const securityInput = secConfig == null ? {} : { apiKey: secConfig };
  const requestSecurity = resolveGlobalSecurity(securityInput);

  const context = {
    options: client._options,
    baseURL: options?.serverURL ?? client._baseURL ?? "",
    operationID:
      "generate_index_summary_v1_rag_indexes_index__index_id__summary_field__language__post",
    oAuth2Scopes: null,

    resolvedSecurity: requestSecurity,

    securitySource: client._options.apiKey,
    retryConfig: options?.retries
      || client._options.retryConfig
      || { strategy: "none" },
    retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
  };

  const requestRes = client._createRequest(context, {
    security: requestSecurity,
    method: "POST",
    baseURL: options?.serverURL,
    path: path,
    headers: headers,
    body: body,
    userAgent: client._options.userAgent,
    timeoutMs: options?.timeoutMs || client._options.timeoutMs || 300000,
  }, options);
  if (!requestRes.ok) {
    return [requestRes, { status: "invalid" }];
  }
  const req = requestRes.value;

  const doResult = await client._do(req, {
    context,
    isErrorStatusCode: (statusCode: number) =>
      matchStatusCode({ status: statusCode } as Response, ["4XX", "5XX"]),
    retryConfig: context.retryConfig,
    retryCodes: context.retryCodes,
  });
  if (!doResult.ok) {
    return [doResult, { status: "request-error", request: req }];
  }
  const response = doResult.value;

  const responseFields = {
    HttpMeta: { Response: response, Request: req },
  };

  const [result] = await M.match<
    JsonLStream<
      operations.GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes
    >,
    | errors.HTTPValidationError
    | MistralError
    | ResponseValidationError
    | ConnectionError
    | RequestAbortedError
    | RequestTimeoutError
    | InvalidRequestError
    | UnexpectedClientError
    | SDKValidationError
  >(
    M.jsonl(
      200,
      z.custom<ReadableStream<Uint8Array>>(x => x instanceof ReadableStream)
        .transform(stream => {
          return new JsonLStream({
            stream,
            decoder(rawEvent) {
              const schema =
                operations
                  .GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes$inboundSchema;
              return schema.parse(rawEvent);
            },
          });
        }),
      { ctype: "application/x-ndjson" },
    ),
    M.jsonErr(422, errors.HTTPValidationError$inboundSchema),
    M.fail([400, 403, 404, "4XX"]),
    M.fail([500, "5XX"]),
  )(response, req, { extraFields: responseFields });
  if (!result.ok) {
    return [result, { status: "complete", request: req, response }];
  }

  return [result, { status: "complete", request: req, response }];
}
