// @mistralai/mistralai/src/models/operations/connectordeleteallusercredentialsv1.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 5e56364d4bee
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";

export type ConnectorDeleteAllUserCredentialsV1Request = {
  connectorIdOrName: string;
};

/** @internal */
export type ConnectorDeleteAllUserCredentialsV1Request$Outbound = {
  connector_id_or_name: string;
};

/** @internal */
export const ConnectorDeleteAllUserCredentialsV1Request$outboundSchema:
  z.ZodType<
    ConnectorDeleteAllUserCredentialsV1Request$Outbound,
    ConnectorDeleteAllUserCredentialsV1Request
  > = z.object({
    connectorIdOrName: z.string(),
  }).transform((v) => {
    return remap$(v, {
      connectorIdOrName: "connector_id_or_name",
    });
  });

export function connectorDeleteAllUserCredentialsV1RequestToJSON(
  connectorDeleteAllUserCredentialsV1Request:
    ConnectorDeleteAllUserCredentialsV1Request,
): string {
  return JSON.stringify(
    ConnectorDeleteAllUserCredentialsV1Request$outboundSchema.parse(
      connectorDeleteAllUserCredentialsV1Request,
    ),
  );
}
