// @mistralai/mistralai/src/models/operations/generateindexsummaryv1ragindexesindexindexidsummaryfieldlanguagepost.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 71ab6a31fe57
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { ClosedEnum } from "../../types/enums.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import { smartUnion } from "../../types/smartUnion.js";
import * as components from "../components/index.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage =
  {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
  } as const;
export type GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage =
  ClosedEnum<
    typeof GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage
  >;

export type GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest =
  {
    indexId: string;
    language:
      GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage;
  };

/**
 * Successful Response
 */
export type GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes =
  | components.SummaryMetadata
  | components.SummaryChunk
  | components.SummaryError;

/** @internal */
export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage$outboundSchema:
  z.ZodEnum<
    typeof GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage
  > = z.enum(
    GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage,
  );

/** @internal */
export type GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest$Outbound =
  {
    index_id: string;
    language: string;
  };

/** @internal */
export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest$outboundSchema:
  z.ZodType<
    GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest$Outbound,
    GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest
  > = z.object({
    indexId: z.string(),
    language:
      GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostLanguage$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      indexId: "index_id",
    });
  });

export function generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequestToJSON(
  generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest:
    GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest,
): string {
  return JSON.stringify(
    GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest$outboundSchema
      .parse(
        generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostRequest,
      ),
  );
}

/** @internal */
export const GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes$inboundSchema:
  z.ZodType<
    GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes,
    unknown
  > = smartUnion([
    components.SummaryMetadata$inboundSchema,
    components.SummaryChunk$inboundSchema,
    components.SummaryError$inboundSchema,
  ]);

export function generateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypesFromJSON(
  jsonString: string,
): SafeParseResult<
  GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes,
  SDKValidationError
> {
  return safeParse(
    jsonString,
    (x) =>
      GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes$inboundSchema
        .parse(JSON.parse(x)),
    `Failed to parse 'GenerateIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePostSummaryStreamTypes' from JSON`,
  );
}
