// @mistralai/mistralai/src/models/operations/getindexsummaryv1ragindexesindexindexidsummaryfieldlanguageget.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 0b8d6ed58e09
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { ClosedEnum } from "../../types/enums.js";

export const GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage =
  {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
  } as const;
export type GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage =
  ClosedEnum<
    typeof GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage
  >;

export type GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest =
  {
    indexId: string;
    language:
      GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage;
  };

/** @internal */
export const GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage$outboundSchema:
  z.ZodEnum<
    typeof GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage
  > = z.enum(
    GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage,
  );

/** @internal */
export type GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest$Outbound =
  {
    index_id: string;
    language: string;
  };

/** @internal */
export const GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest$outboundSchema:
  z.ZodType<
    GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest$Outbound,
    GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest
  > = z.object({
    indexId: z.string(),
    language:
      GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetLanguage$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      indexId: "index_id",
    });
  });

export function getIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequestToJSON(
  getIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest:
    GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest,
): string {
  return JSON.stringify(
    GetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest$outboundSchema
      .parse(
        getIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguageGetRequest,
      ),
  );
}
