// @mistralai/mistralai/src/models/operations/promptsgetversion.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 8027286369ad
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";

export type PromptsGetVersionRequest = {
  promptId: string;
  version: number;
  fields?: Array<string> | undefined;
};

/** @internal */
export type PromptsGetVersionRequest$Outbound = {
  prompt_id: string;
  version: number;
  fields?: Array<string> | undefined;
};

/** @internal */
export const PromptsGetVersionRequest$outboundSchema: z.ZodType<
  PromptsGetVersionRequest$Outbound,
  PromptsGetVersionRequest
> = z.object({
  promptId: z.string(),
  version: z.int(),
  fields: z.array(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    promptId: "prompt_id",
  });
});

export function promptsGetVersionRequestToJSON(
  promptsGetVersionRequest: PromptsGetVersionRequest,
): string {
  return JSON.stringify(
    PromptsGetVersionRequest$outboundSchema.parse(promptsGetVersionRequest),
  );
}
