// @mistralai/mistralai/src/models/operations/promptsupdateversionmetadata.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: f6b0b69eb311
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type UpdatePromptVersionRequest = {
  /**
   * Notes for this version.
   */
  notes?: string | null | undefined;
  /**
   * Presence wrapper for a set of alias labels on update RPCs. As a message field it carries presence, so callers can distinguish "leave aliases unchanged" (field omitted) from "clear all aliases" (field set, empty ``values``).
   */
  aliases?: components.AliasList | undefined;
};

export type PromptsUpdateVersionMetadataRequest = {
  promptId: string;
  version: number;
  requestBody: UpdatePromptVersionRequest;
};

/** @internal */
export type UpdatePromptVersionRequest$Outbound = {
  notes?: string | null | undefined;
  aliases?: components.AliasList$Outbound | undefined;
};

/** @internal */
export const UpdatePromptVersionRequest$outboundSchema: z.ZodType<
  UpdatePromptVersionRequest$Outbound,
  UpdatePromptVersionRequest
> = z.object({
  notes: z.nullable(z.string()).optional(),
  aliases: components.AliasList$outboundSchema.optional(),
});

export function updatePromptVersionRequestToJSON(
  updatePromptVersionRequest: UpdatePromptVersionRequest,
): string {
  return JSON.stringify(
    UpdatePromptVersionRequest$outboundSchema.parse(updatePromptVersionRequest),
  );
}

/** @internal */
export type PromptsUpdateVersionMetadataRequest$Outbound = {
  prompt_id: string;
  version: number;
  RequestBody: UpdatePromptVersionRequest$Outbound;
};

/** @internal */
export const PromptsUpdateVersionMetadataRequest$outboundSchema: z.ZodType<
  PromptsUpdateVersionMetadataRequest$Outbound,
  PromptsUpdateVersionMetadataRequest
> = z.object({
  promptId: z.string(),
  version: z.int(),
  requestBody: z.lazy(() => UpdatePromptVersionRequest$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    promptId: "prompt_id",
    requestBody: "RequestBody",
  });
});

export function promptsUpdateVersionMetadataRequestToJSON(
  promptsUpdateVersionMetadataRequest: PromptsUpdateVersionMetadataRequest,
): string {
  return JSON.stringify(
    PromptsUpdateVersionMetadataRequest$outboundSchema.parse(
      promptsUpdateVersionMetadataRequest,
    ),
  );
}
