// @mistralai/mistralai/src/models/operations/promptsupdate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: cfdc043733bf
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type UpdatePromptRequest = {
  /**
   * Display title.
   */
  title?: string | null | undefined;
  /**
   * Display description.
   */
  description?: string | null | undefined;
  sharingScope?: components.RegistrySharingScope | undefined;
};

export type PromptsUpdateRequest = {
  promptId: string;
  requestBody: UpdatePromptRequest;
};

/** @internal */
export type UpdatePromptRequest$Outbound = {
  title?: string | null | undefined;
  description?: string | null | undefined;
  sharingScope?: string | undefined;
};

/** @internal */
export const UpdatePromptRequest$outboundSchema: z.ZodType<
  UpdatePromptRequest$Outbound,
  UpdatePromptRequest
> = z.object({
  title: z.nullable(z.string()).optional(),
  description: z.nullable(z.string()).optional(),
  sharingScope: components.RegistrySharingScope$outboundSchema.optional(),
});

export function updatePromptRequestToJSON(
  updatePromptRequest: UpdatePromptRequest,
): string {
  return JSON.stringify(
    UpdatePromptRequest$outboundSchema.parse(updatePromptRequest),
  );
}

/** @internal */
export type PromptsUpdateRequest$Outbound = {
  prompt_id: string;
  RequestBody: UpdatePromptRequest$Outbound;
};

/** @internal */
export const PromptsUpdateRequest$outboundSchema: z.ZodType<
  PromptsUpdateRequest$Outbound,
  PromptsUpdateRequest
> = z.object({
  promptId: z.string(),
  requestBody: z.lazy(() => UpdatePromptRequest$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    promptId: "prompt_id",
    requestBody: "RequestBody",
  });
});

export function promptsUpdateRequestToJSON(
  promptsUpdateRequest: PromptsUpdateRequest,
): string {
  return JSON.stringify(
    PromptsUpdateRequest$outboundSchema.parse(promptsUpdateRequest),
  );
}
