// @mistralai/mistralai/src/models/operations/promptsget.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1f6e697e1803
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";

export type PromptsGetRequest = {
  promptId: string;
  version?: number | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

/** @internal */
export type PromptsGetRequest$Outbound = {
  prompt_id: string;
  version?: number | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

/** @internal */
export const PromptsGetRequest$outboundSchema: z.ZodType<
  PromptsGetRequest$Outbound,
  PromptsGetRequest
> = z.object({
  promptId: z.string(),
  version: z.int().optional(),
  alias: z.string().optional(),
  fields: z.array(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    promptId: "prompt_id",
  });
});

export function promptsGetRequestToJSON(
  promptsGetRequest: PromptsGetRequest,
): string {
  return JSON.stringify(
    PromptsGetRequest$outboundSchema.parse(promptsGetRequest),
  );
}
