// @mistralai/mistralai/src/models/operations/promptslist.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 7e6054cd6360
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import * as components from "../components/index.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type PromptsListRequest = {
  pageSize?: number | undefined;
  pageToken?: string | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

export type PromptsListResponse = {
  result: components.ListPromptsResponse;
};

/** @internal */
export type PromptsListRequest$Outbound = {
  pageSize?: number | undefined;
  pageToken?: string | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

/** @internal */
export const PromptsListRequest$outboundSchema: z.ZodType<
  PromptsListRequest$Outbound,
  PromptsListRequest
> = z.object({
  pageSize: z.int().optional(),
  pageToken: z.string().optional(),
  alias: z.string().optional(),
  fields: z.array(z.string()).optional(),
});

export function promptsListRequestToJSON(
  promptsListRequest: PromptsListRequest,
): string {
  return JSON.stringify(
    PromptsListRequest$outboundSchema.parse(promptsListRequest),
  );
}

/** @internal */
export const PromptsListResponse$inboundSchema: z.ZodType<
  PromptsListResponse,
  unknown
> = z.object({
  Result: components.ListPromptsResponse$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "Result": "result",
  });
});

export function promptsListResponseFromJSON(
  jsonString: string,
): SafeParseResult<PromptsListResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => PromptsListResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'PromptsListResponse' from JSON`,
  );
}
