// @mistralai/mistralai/src/models/operations/setindexsummaryv1ragindexesindexindexidsummaryfieldlanguageput.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 053edbeba171
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { ClosedEnum } from "../../types/enums.js";
import * as components from "../components/index.js";

export const SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage =
  {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
  } as const;
export type SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage =
  ClosedEnum<
    typeof SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage
  >;

export type SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest =
  {
    indexId: string;
    language:
      SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage;
    updateSummaryRequestSummary: components.UpdateSummaryRequestSummary;
  };

/** @internal */
export const SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage$outboundSchema:
  z.ZodEnum<
    typeof SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage
  > = z.enum(
    SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage,
  );

/** @internal */
export type SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest$Outbound =
  {
    index_id: string;
    language: string;
    UpdateSummaryRequestSummary:
      components.UpdateSummaryRequestSummary$Outbound;
  };

/** @internal */
export const SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest$outboundSchema:
  z.ZodType<
    SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest$Outbound,
    SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest
  > = z.object({
    indexId: z.string(),
    language:
      SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutLanguage$outboundSchema,
    updateSummaryRequestSummary:
      components.UpdateSummaryRequestSummary$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      indexId: "index_id",
      updateSummaryRequestSummary: "UpdateSummaryRequestSummary",
    });
  });

export function setIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequestToJSON(
  setIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest:
    SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest,
): string {
  return JSON.stringify(
    SetIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest$outboundSchema
      .parse(
        setIndexSummaryV1RagIndexesIndexIndexIdSummaryFieldLanguagePutRequest,
      ),
  );
}
