// @mistralai/mistralai/src/models/operations/setschemasummaryv1ragindexesindexindexidschemasschemaschemaidsummaryfieldlanguageput.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 1359b3081a8e
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { ClosedEnum } from "../../types/enums.js";
import * as components from "../components/index.js";

export const SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage =
  {
    En: "en",
    Fr: "fr",
    Es: "es",
    De: "de",
    It: "it",
    PtBr: "pt_br",
    Pl: "pl",
    Ar: "ar",
    Nl: "nl",
  } as const;
export type SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage =
  ClosedEnum<
    typeof SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage
  >;

export type SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest =
  {
    indexId: string;
    schemaId: string;
    language:
      SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage;
    updateSummaryRequestSummary: components.UpdateSummaryRequestSummary;
  };

/** @internal */
export const SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage$outboundSchema:
  z.ZodEnum<
    typeof SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage
  > = z.enum(
    SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage,
  );

/** @internal */
export type SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest$Outbound =
  {
    index_id: string;
    schema_id: string;
    language: string;
    UpdateSummaryRequestSummary:
      components.UpdateSummaryRequestSummary$Outbound;
  };

/** @internal */
export const SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest$outboundSchema:
  z.ZodType<
    SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest$Outbound,
    SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest
  > = z.object({
    indexId: z.string(),
    schemaId: z.string(),
    language:
      SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutLanguage$outboundSchema,
    updateSummaryRequestSummary:
      components.UpdateSummaryRequestSummary$outboundSchema,
  }).transform((v) => {
    return remap$(v, {
      indexId: "index_id",
      schemaId: "schema_id",
      updateSummaryRequestSummary: "UpdateSummaryRequestSummary",
    });
  });

export function setSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequestToJSON(
  setSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest:
    SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest,
): string {
  return JSON.stringify(
    SetSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest$outboundSchema
      .parse(
        setSchemaSummaryV1RagIndexesIndexIndexIdSchemasSchemaSchemaIdSummaryFieldLanguagePutRequest,
      ),
  );
}
