// @mistralai/mistralai/src/models/operations/skillsget.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 227d3b4dad4c
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";

export type SkillsGetRequest = {
  skillId: string;
  version?: number | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

/** @internal */
export type SkillsGetRequest$Outbound = {
  skill_id: string;
  version?: number | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

/** @internal */
export const SkillsGetRequest$outboundSchema: z.ZodType<
  SkillsGetRequest$Outbound,
  SkillsGetRequest
> = z.object({
  skillId: z.string(),
  version: z.int().optional(),
  alias: z.string().optional(),
  fields: z.array(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    skillId: "skill_id",
  });
});

export function skillsGetRequestToJSON(
  skillsGetRequest: SkillsGetRequest,
): string {
  return JSON.stringify(
    SkillsGetRequest$outboundSchema.parse(skillsGetRequest),
  );
}
