// @mistralai/mistralai/src/models/operations/skillsgetversion.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 912ee1c035b6
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";

export type SkillsGetVersionRequest = {
  skillId: string;
  version: number;
  fields?: Array<string> | undefined;
};

/** @internal */
export type SkillsGetVersionRequest$Outbound = {
  skill_id: string;
  version: number;
  fields?: Array<string> | undefined;
};

/** @internal */
export const SkillsGetVersionRequest$outboundSchema: z.ZodType<
  SkillsGetVersionRequest$Outbound,
  SkillsGetVersionRequest
> = z.object({
  skillId: z.string(),
  version: z.int(),
  fields: z.array(z.string()).optional(),
}).transform((v) => {
  return remap$(v, {
    skillId: "skill_id",
  });
});

export function skillsGetVersionRequestToJSON(
  skillsGetVersionRequest: SkillsGetVersionRequest,
): string {
  return JSON.stringify(
    SkillsGetVersionRequest$outboundSchema.parse(skillsGetVersionRequest),
  );
}
