// @mistralai/mistralai/src/models/operations/skillsupdate.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3511172f0c03
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type UpdateSkillRequest = {
  sharingScope?: components.RegistrySharingScope | undefined;
};

export type SkillsUpdateRequest = {
  skillId: string;
  requestBody: UpdateSkillRequest;
};

/** @internal */
export type UpdateSkillRequest$Outbound = {
  sharingScope?: string | undefined;
};

/** @internal */
export const UpdateSkillRequest$outboundSchema: z.ZodType<
  UpdateSkillRequest$Outbound,
  UpdateSkillRequest
> = z.object({
  sharingScope: components.RegistrySharingScope$outboundSchema.optional(),
});

export function updateSkillRequestToJSON(
  updateSkillRequest: UpdateSkillRequest,
): string {
  return JSON.stringify(
    UpdateSkillRequest$outboundSchema.parse(updateSkillRequest),
  );
}

/** @internal */
export type SkillsUpdateRequest$Outbound = {
  skill_id: string;
  RequestBody: UpdateSkillRequest$Outbound;
};

/** @internal */
export const SkillsUpdateRequest$outboundSchema: z.ZodType<
  SkillsUpdateRequest$Outbound,
  SkillsUpdateRequest
> = z.object({
  skillId: z.string(),
  requestBody: z.lazy(() => UpdateSkillRequest$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    skillId: "skill_id",
    requestBody: "RequestBody",
  });
});

export function skillsUpdateRequestToJSON(
  skillsUpdateRequest: SkillsUpdateRequest,
): string {
  return JSON.stringify(
    SkillsUpdateRequest$outboundSchema.parse(skillsUpdateRequest),
  );
}
