// @mistralai/mistralai/src/models/operations/skillsupdateversionmetadata.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e11580c09cf3
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type UpdateSkillVersionRequest = {
  /**
   * Notes for this version.
   */
  notes?: string | null | undefined;
  /**
   * Presence wrapper for a set of alias labels on update RPCs. As a message field it carries presence, so callers can distinguish "leave aliases unchanged" (field omitted) from "clear all aliases" (field set, empty ``values``).
   */
  aliases?: components.AliasList | undefined;
};

export type SkillsUpdateVersionMetadataRequest = {
  skillId: string;
  version: number;
  requestBody: UpdateSkillVersionRequest;
};

/** @internal */
export type UpdateSkillVersionRequest$Outbound = {
  notes?: string | null | undefined;
  aliases?: components.AliasList$Outbound | undefined;
};

/** @internal */
export const UpdateSkillVersionRequest$outboundSchema: z.ZodType<
  UpdateSkillVersionRequest$Outbound,
  UpdateSkillVersionRequest
> = z.object({
  notes: z.nullable(z.string()).optional(),
  aliases: components.AliasList$outboundSchema.optional(),
});

export function updateSkillVersionRequestToJSON(
  updateSkillVersionRequest: UpdateSkillVersionRequest,
): string {
  return JSON.stringify(
    UpdateSkillVersionRequest$outboundSchema.parse(updateSkillVersionRequest),
  );
}

/** @internal */
export type SkillsUpdateVersionMetadataRequest$Outbound = {
  skill_id: string;
  version: number;
  RequestBody: UpdateSkillVersionRequest$Outbound;
};

/** @internal */
export const SkillsUpdateVersionMetadataRequest$outboundSchema: z.ZodType<
  SkillsUpdateVersionMetadataRequest$Outbound,
  SkillsUpdateVersionMetadataRequest
> = z.object({
  skillId: z.string(),
  version: z.int(),
  requestBody: z.lazy(() => UpdateSkillVersionRequest$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    skillId: "skill_id",
    requestBody: "RequestBody",
  });
});

export function skillsUpdateVersionMetadataRequestToJSON(
  skillsUpdateVersionMetadataRequest: SkillsUpdateVersionMetadataRequest,
): string {
  return JSON.stringify(
    SkillsUpdateVersionMetadataRequest$outboundSchema.parse(
      skillsUpdateVersionMetadataRequest,
    ),
  );
}
