// @mistralai/mistralai/src/models/operations/skillslist.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 3eff96e8881c
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
import { Result as SafeParseResult } from "../../types/fp.js";
import * as components from "../components/index.js";
import { SDKValidationError } from "../errors/sdkvalidationerror.js";

export type SkillsListRequest = {
  pageSize?: number | undefined;
  pageToken?: string | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

export type SkillsListResponse = {
  result: components.ListSkillsResponse;
};

/** @internal */
export type SkillsListRequest$Outbound = {
  pageSize?: number | undefined;
  pageToken?: string | undefined;
  alias?: string | undefined;
  fields?: Array<string> | undefined;
};

/** @internal */
export const SkillsListRequest$outboundSchema: z.ZodType<
  SkillsListRequest$Outbound,
  SkillsListRequest
> = z.object({
  pageSize: z.int().optional(),
  pageToken: z.string().optional(),
  alias: z.string().optional(),
  fields: z.array(z.string()).optional(),
});

export function skillsListRequestToJSON(
  skillsListRequest: SkillsListRequest,
): string {
  return JSON.stringify(
    SkillsListRequest$outboundSchema.parse(skillsListRequest),
  );
}

/** @internal */
export const SkillsListResponse$inboundSchema: z.ZodType<
  SkillsListResponse,
  unknown
> = z.object({
  Result: components.ListSkillsResponse$inboundSchema,
}).transform((v) => {
  return remap$(v, {
    "Result": "result",
  });
});

export function skillsListResponseFromJSON(
  jsonString: string,
): SafeParseResult<SkillsListResponse, SDKValidationError> {
  return safeParse(
    jsonString,
    (x) => SkillsListResponse$inboundSchema.parse(JSON.parse(x)),
    `Failed to parse 'SkillsListResponse' from JSON`,
  );
}
