// @mistralai/mistralai/src/models/operations/skillscreateversion.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: e89ac7211d41
 */

import * as z from "zod/v4";
import { remap as remap$ } from "../../lib/primitives.js";
import * as components from "../components/index.js";

export type CreateSkillVersionRequest = {
  /**
   * Versioned skill content.
   */
  definition: components.SkillDefinition;
  /**
   * Notes for this version.
   */
  notes?: string | null | undefined;
  /**
   * Aliases pointing to this version.
   */
  aliases?: Array<string> | undefined;
};

export type SkillsCreateVersionRequest = {
  skillId: string;
  requestBody: CreateSkillVersionRequest;
};

/** @internal */
export type CreateSkillVersionRequest$Outbound = {
  definition: components.SkillDefinition$Outbound;
  notes?: string | null | undefined;
  aliases?: Array<string> | undefined;
};

/** @internal */
export const CreateSkillVersionRequest$outboundSchema: z.ZodType<
  CreateSkillVersionRequest$Outbound,
  CreateSkillVersionRequest
> = z.object({
  definition: components.SkillDefinition$outboundSchema,
  notes: z.nullable(z.string()).optional(),
  aliases: z.array(z.string()).optional(),
});

export function createSkillVersionRequestToJSON(
  createSkillVersionRequest: CreateSkillVersionRequest,
): string {
  return JSON.stringify(
    CreateSkillVersionRequest$outboundSchema.parse(createSkillVersionRequest),
  );
}

/** @internal */
export type SkillsCreateVersionRequest$Outbound = {
  skill_id: string;
  RequestBody: CreateSkillVersionRequest$Outbound;
};

/** @internal */
export const SkillsCreateVersionRequest$outboundSchema: z.ZodType<
  SkillsCreateVersionRequest$Outbound,
  SkillsCreateVersionRequest
> = z.object({
  skillId: z.string(),
  requestBody: z.lazy(() => CreateSkillVersionRequest$outboundSchema),
}).transform((v) => {
  return remap$(v, {
    skillId: "skill_id",
    requestBody: "RequestBody",
  });
});

export function skillsCreateVersionRequestToJSON(
  skillsCreateVersionRequest: SkillsCreateVersionRequest,
): string {
  return JSON.stringify(
    SkillsCreateVersionRequest$outboundSchema.parse(skillsCreateVersionRequest),
  );
}
