// @mistralai/mistralai/esm/models/components/prompt.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: bffa6c75bb04
 */
import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { PromptDefinition$inboundSchema, } from "./promptdefinition.js";
import { RegistrySharingScope$inboundSchema, } from "./registrysharingscope.js";
/** @internal */
export const Prompt$inboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    definition: PromptDefinition$inboundSchema.optional(),
    version: z.int().optional(),
    notes: z.string().optional(),
    aliases: z.array(z.string()).optional(),
    sharingScope: RegistrySharingScope$inboundSchema.optional(),
    createdAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    updatedAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    latestVersion: z.int().optional(),
    title: z.string().optional(),
    description: z.string().optional(),
});
export function promptFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Prompt$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Prompt' from JSON`);
}
