// @mistralai/mistralai/esm/models/components/skillassetcontent.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 6f35b94388b6
 */
import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { smartUnion } from "../../types/smartUnion.js";
/** @internal */
export const SkillAssetContentTextContent$inboundSchema = z.object({
    textContent: z.string(),
    isExecutable: z.boolean().optional(),
});
/** @internal */
export const SkillAssetContentTextContent$outboundSchema = z.object({
    textContent: z.string(),
    isExecutable: z.boolean().optional(),
});
export function skillAssetContentTextContentToJSON(skillAssetContentTextContent) {
    return JSON.stringify(SkillAssetContentTextContent$outboundSchema.parse(skillAssetContentTextContent));
}
export function skillAssetContentTextContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SkillAssetContentTextContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SkillAssetContentTextContent' from JSON`);
}
/** @internal */
export const RawContent$inboundSchema = z
    .object({
    rawContent: z.string(),
    isExecutable: z.boolean().optional(),
});
/** @internal */
export const RawContent$outboundSchema = z.object({
    rawContent: z.string(),
    isExecutable: z.boolean().optional(),
});
export function rawContentToJSON(rawContent) {
    return JSON.stringify(RawContent$outboundSchema.parse(rawContent));
}
export function rawContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RawContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RawContent' from JSON`);
}
/** @internal */
export const SkillAssetContent$inboundSchema = smartUnion([
    z.lazy(() => RawContent$inboundSchema),
    z.lazy(() => SkillAssetContentTextContent$inboundSchema),
]);
/** @internal */
export const SkillAssetContent$outboundSchema = smartUnion([
    z.lazy(() => RawContent$outboundSchema),
    z.lazy(() => SkillAssetContentTextContent$outboundSchema),
]);
export function skillAssetContentToJSON(skillAssetContent) {
    return JSON.stringify(SkillAssetContent$outboundSchema.parse(skillAssetContent));
}
export function skillAssetContentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SkillAssetContent$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SkillAssetContent' from JSON`);
}
