// @mistralai/mistralai/esm/models/components/skilldefinition.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 4e9a30e817bb
 */
import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { SkillAssetContent$inboundSchema, SkillAssetContent$outboundSchema, } from "./skillassetcontent.js";
/** @internal */
export const SkillDefinition$inboundSchema = z.object({
    description: z.string().optional(),
    body: z.string().optional(),
    assets: z.record(z.string(), SkillAssetContent$inboundSchema).optional(),
});
/** @internal */
export const SkillDefinition$outboundSchema = z.object({
    description: z.string().optional(),
    body: z.string().optional(),
    assets: z.record(z.string(), SkillAssetContent$outboundSchema).optional(),
});
export function skillDefinitionToJSON(skillDefinition) {
    return JSON.stringify(SkillDefinition$outboundSchema.parse(skillDefinition));
}
export function skillDefinitionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SkillDefinition$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SkillDefinition' from JSON`);
}
