// @mistralai/mistralai/esm/models/components/skill.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 762b25305020
 */
import * as z from "zod/v4";
import { safeParse } from "../../lib/schemas.js";
import { RegistrySharingScope$inboundSchema, } from "./registrysharingscope.js";
import { SkillDefinition$inboundSchema, } from "./skilldefinition.js";
/** @internal */
export const Skill$inboundSchema = z.object({
    id: z.string().optional(),
    name: z.string().optional(),
    definition: SkillDefinition$inboundSchema.optional(),
    version: z.int().optional(),
    notes: z.string().optional(),
    aliases: z.array(z.string()).optional(),
    sharingScope: RegistrySharingScope$inboundSchema.optional(),
    createdAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    updatedAt: z.iso.datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    latestVersion: z.int().optional(),
});
export function skillFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Skill$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Skill' from JSON`);
}
