// @mistralai/mistralai/src/sdk/prompts.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: a20685c0ab9f
 */

import { betaPromptsCreate } from "../funcs/betaPromptsCreate.js";
import { betaPromptsCreateVersion } from "../funcs/betaPromptsCreateVersion.js";
import { betaPromptsDelete } from "../funcs/betaPromptsDelete.js";
import { betaPromptsGet } from "../funcs/betaPromptsGet.js";
import { betaPromptsGetVersion } from "../funcs/betaPromptsGetVersion.js";
import { betaPromptsList } from "../funcs/betaPromptsList.js";
import { betaPromptsListVersions } from "../funcs/betaPromptsListVersions.js";
import { betaPromptsUpdateMetadata } from "../funcs/betaPromptsUpdateMetadata.js";
import { betaPromptsUpdateVersionMetadata } from "../funcs/betaPromptsUpdateVersionMetadata.js";
import { ClientSDK, RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { PageIterator, unwrapResultIterator } from "../types/operations.js";

export class Prompts extends ClientSDK {
  /**
   * ListPrompts
   */
  async list(
    request?: operations.PromptsListRequest | undefined,
    options?: RequestOptions,
  ): Promise<PageIterator<operations.PromptsListResponse, { cursor: string }>> {
    return unwrapResultIterator(betaPromptsList(
      this,
      request,
      options,
    ));
  }

  /**
   * CreatePrompt
   */
  async create(
    request: components.CreatePromptRequest,
    options?: RequestOptions,
  ): Promise<components.Prompt> {
    return unwrapAsync(betaPromptsCreate(
      this,
      request,
      options,
    ));
  }

  /**
   * GetPrompt
   */
  async get(
    request: operations.PromptsGetRequest,
    options?: RequestOptions,
  ): Promise<components.Prompt> {
    return unwrapAsync(betaPromptsGet(
      this,
      request,
      options,
    ));
  }

  /**
   * DeletePrompt
   */
  async delete(
    request: operations.PromptsDeleteRequest,
    options?: RequestOptions,
  ): Promise<components.DeletePromptResponse> {
    return unwrapAsync(betaPromptsDelete(
      this,
      request,
      options,
    ));
  }

  /**
   * UpdatePrompt
   */
  async updateMetadata(
    promptId: string,
    requestBody: operations.UpdatePromptRequest,
    options?: RequestOptions,
  ): Promise<components.Prompt> {
    return unwrapAsync(betaPromptsUpdateMetadata(
      this,
      promptId,
      requestBody,
      options,
    ));
  }

  /**
   * ListPromptVersions
   */
  async listVersions(
    request: operations.PromptsListVersionsRequest,
    options?: RequestOptions,
  ): Promise<components.ListPromptVersionsResponse> {
    return unwrapAsync(betaPromptsListVersions(
      this,
      request,
      options,
    ));
  }

  /**
   * CreatePromptVersion
   */
  async createVersion(
    promptId: string,
    requestBody: operations.CreatePromptVersionRequest,
    options?: RequestOptions,
  ): Promise<components.CreatePromptVersionResponse> {
    return unwrapAsync(betaPromptsCreateVersion(
      this,
      promptId,
      requestBody,
      options,
    ));
  }

  /**
   * GetPromptVersion
   */
  async getVersion(
    request: operations.PromptsGetVersionRequest,
    options?: RequestOptions,
  ): Promise<components.Prompt> {
    return unwrapAsync(betaPromptsGetVersion(
      this,
      request,
      options,
    ));
  }

  /**
   * UpdatePromptVersionMetadata
   */
  async updateVersionMetadata(
    promptId: string,
    version: number,
    requestBody: operations.UpdatePromptVersionRequest,
    options?: RequestOptions,
  ): Promise<components.Prompt> {
    return unwrapAsync(betaPromptsUpdateVersionMetadata(
      this,
      promptId,
      version,
      requestBody,
      options,
    ));
  }
}
