// @mistralai/mistralai/src/sdk/skills.ts
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 * @generated-id: 847667a72f83
 */

import { betaSkillsCreate } from "../funcs/betaSkillsCreate.js";
import { betaSkillsCreateVersion } from "../funcs/betaSkillsCreateVersion.js";
import { betaSkillsDelete } from "../funcs/betaSkillsDelete.js";
import { betaSkillsGet } from "../funcs/betaSkillsGet.js";
import { betaSkillsGetVersion } from "../funcs/betaSkillsGetVersion.js";
import { betaSkillsList } from "../funcs/betaSkillsList.js";
import { betaSkillsListVersions } from "../funcs/betaSkillsListVersions.js";
import { betaSkillsUpdateMetadata } from "../funcs/betaSkillsUpdateMetadata.js";
import { betaSkillsUpdateVersionMetadata } from "../funcs/betaSkillsUpdateVersionMetadata.js";
import { ClientSDK, RequestOptions } from "../lib/sdks.js";
import * as components from "../models/components/index.js";
import * as operations from "../models/operations/index.js";
import { unwrapAsync } from "../types/fp.js";
import { PageIterator, unwrapResultIterator } from "../types/operations.js";

export class Skills extends ClientSDK {
  /**
   * ListSkills
   */
  async list(
    request?: operations.SkillsListRequest | undefined,
    options?: RequestOptions,
  ): Promise<PageIterator<operations.SkillsListResponse, { cursor: string }>> {
    return unwrapResultIterator(betaSkillsList(
      this,
      request,
      options,
    ));
  }

  /**
   * CreateSkill
   */
  async create(
    request: components.CreateSkillRequest,
    options?: RequestOptions,
  ): Promise<components.Skill> {
    return unwrapAsync(betaSkillsCreate(
      this,
      request,
      options,
    ));
  }

  /**
   * GetSkill
   */
  async get(
    request: operations.SkillsGetRequest,
    options?: RequestOptions,
  ): Promise<components.Skill> {
    return unwrapAsync(betaSkillsGet(
      this,
      request,
      options,
    ));
  }

  /**
   * DeleteSkill
   */
  async delete(
    request: operations.SkillsDeleteRequest,
    options?: RequestOptions,
  ): Promise<components.DeleteSkillResponse> {
    return unwrapAsync(betaSkillsDelete(
      this,
      request,
      options,
    ));
  }

  /**
   * UpdateSkill
   */
  async updateMetadata(
    skillId: string,
    requestBody: operations.UpdateSkillRequest,
    options?: RequestOptions,
  ): Promise<components.Skill> {
    return unwrapAsync(betaSkillsUpdateMetadata(
      this,
      skillId,
      requestBody,
      options,
    ));
  }

  /**
   * ListSkillVersions
   */
  async listVersions(
    request: operations.SkillsListVersionsRequest,
    options?: RequestOptions,
  ): Promise<components.ListSkillVersionsResponse> {
    return unwrapAsync(betaSkillsListVersions(
      this,
      request,
      options,
    ));
  }

  /**
   * CreateSkillVersion
   */
  async createVersion(
    skillId: string,
    requestBody: operations.CreateSkillVersionRequest,
    options?: RequestOptions,
  ): Promise<components.CreateSkillVersionResponse> {
    return unwrapAsync(betaSkillsCreateVersion(
      this,
      skillId,
      requestBody,
      options,
    ));
  }

  /**
   * GetSkillVersion
   */
  async getVersion(
    request: operations.SkillsGetVersionRequest,
    options?: RequestOptions,
  ): Promise<components.Skill> {
    return unwrapAsync(betaSkillsGetVersion(
      this,
      request,
      options,
    ));
  }

  /**
   * UpdateSkillVersionMetadata
   */
  async updateVersionMetadata(
    skillId: string,
    version: number,
    requestBody: operations.UpdateSkillVersionRequest,
    options?: RequestOptions,
  ): Promise<components.Skill> {
    return unwrapAsync(betaSkillsUpdateVersionMetadata(
      this,
      skillId,
      version,
      requestBody,
      options,
    ));
  }
}
